data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/WithKeysProcessorConfig.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.mutatestring;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+public abstract class WithKeysProcessorConfig implements StringProcessorConfig<String> {
+    @NotEmpty
+    @NotNull
+    @JsonProperty("with_keys")
+    private List<String> withKeys;
+
+    @Override
+    public List<String> getIterativeConfig() {
+        return withKeys;
+    }
+
+    public List<String> getWithKeys() {
+        return withKeys;
+    }
```

Comment:
It looks like these two getters are redundant.

Reply:
They are but I want users to have access to the keys naturally without having to dig in and figure out that `getIterativeConfig` is the keeper of the keys.